apps/web/src/components/Messages/Composer.tsx

```diff
@@ -58,27 +126,77 @@ const Composer: FC<ComposerProps> = ({ sendMessage, conversationKey, disabledInp
     }
   };
 
+  const onAttachmentChange = async (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files?.length) {
+      const file = e.target.files[0];
+
+      const fileReader = new FileReader();
+      fileReader.addEventListener('load', async function () {
+        const data = fileReader.result;
+
+        if (!(data instanceof ArrayBuffer)) {
+          return;
+        }
+
+        const attachment: Attachment = {
+          filename: file.name,
+          mimeType: file.type,
+          data: new Uint8Array(data)
+        };
+
+        setAttachment(attachment);
+      });
+
+      fileReader.readAsArrayBuffer(file);
+    } else {
+      setAttachment(null);
+    }
+  };
+
+  function onDismiss() {
+    setAttachment(null);
+
+    const el = fileInputRef?.current;
```

Comment:
```suggestion
    const el = fileInputRef.current;
```
all reference will always have `current`?